parsl/monitoring/db_manager.py

```diff
@@ -310,40 +341,69 @@ def start(self, priority_queue, node_queue, resource_queue) -> None:
                                          messages=[msg])
                             self.workflow_end = True
 
-                    else:                             # TASK_INFO message
+                    elif msg_type.value == MessageType.TASK_INFO.value:
+                        task_try_id = str(msg['task_id']) + "." + str(msg['try_id'])
                         task_info_all_messages.append(msg)
                         if msg['task_id'] in inserted_tasks:
                             task_info_update_messages.append(msg)
                         else:
                             inserted_tasks.add(msg['task_id'])
                             task_info_insert_messages.append(msg)
 
-                            # check if there is an left_message for this task
-                            if msg['task_id'] in deferred_resource_messages:
+                        try_all_messages.append(msg)
+                        if task_try_id in inserted_tries:
+                            try_update_messages.append(msg)
+                        else:
+                            inserted_tries.add(task_try_id)
+                            try_insert_messages.append(msg)
+
+                            # check if there is a left_message for this task
+                            if task_try_id in deferred_resource_messages:
                                 reprocessable_first_resource_messages.append(
-                                    deferred_resource_messages.pop(msg['task_id']))
+                                    deferred_resource_messages.pop(task_try_id))
+                    else:
+                        raise RuntimeError("Unexpected message type {} received on priority queue".format(msg_type))
 
-                logger.debug(
-                    "Updating and inserting TASK_INFO to all tables")
+                logger.debug("Updating and inserting TASK_INFO to all tables")
+                logger.debug("Updating {} TASK_INFO into workflow table".format(len(task_info_update_messages)))
+                self._update(table=WORKFLOW,
+                             columns=['run_id', 'tasks_failed_count',
+                                      'tasks_completed_count'],
+                             messages=task_info_all_messages)
```

Comment:
This update now processes `task_info_all_messages`, not `task_info_update_messages`, because it should update the workflow table no matter whether the message is an update for an existing task, or a message about a new task.